pkg/scheduler/admission/podnodeconstraints/admission.go

```diff
@@ -0,0 +1,196 @@
+package podnodeconstraints
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	admission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	"github.com/openshift/origin/pkg/authorization/authorizer"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/scheduler/admission/podnodeconstraints/api"
+)
+
+func init() {
+	admission.RegisterPlugin("PodNodeConstraints", func(c clientset.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return NewPodNodeConstraints(pluginConfig), nil
+	})
+}
+
+// NewPodNodeConstraints creates a new admission plugin to prevent objects that contain pod templates
+// from containing node bindings by name or selector based on role permissions.
+func NewPodNodeConstraints(config *api.PodNodeConstraintsConfig) admission.Interface {
+	return &podNodeConstraints{
+		config:  config,
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}
+}
+
+type podNodeConstraints struct {
+	*admission.Handler
+	config     *api.PodNodeConstraintsConfig
+	authorizer authorizer.Authorizer
+}
+
+// resourcesToCheck is a map of resources and corresponding kinds of things that
+// we want handled in this plugin
+var resourcesToCheck = map[unversioned.GroupResource]unversioned.GroupKind{
```

Comment:
Add a TODO to restructure this so that an entry in this map includes the function required to pull the `PodSpec` so that the unit test actually guarantees that this works instead of simply encouraging people to blindly put things in this map.

Remember, you aren't the guy who's going to get called on this.  Someone whose never seen it and is doing a rebase will get tripped up and we want them to handle it correctly.